core-java/src/test/java/com/baeldung/bufferedreader/BufferedReaderExampleUnitTest.java

```diff
@@ -0,0 +1,122 @@
+package com.baeldung.bufferedreader;
+
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
+
+public class BufferedReaderExampleUnitTest {
+
+    public static final String FILE_PATH = "src/main/resources/input.txt";
+
+    @Test
+    public void givenBufferedReader_whenReadAllLines_thenReturnsContent() throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH));
+
+        BufferedReaderExample bre = new BufferedReaderExample();
+        String content = bre.readAllLines(reader);
```

Comment:
It feels a bit odd that some of these are delegated to `BufferedReaderExample` while others aren't. Is there a way that we could do one or the other? Or, is there a reason that it is this way?

Reply:
I'm aware of this problem. Initially all tests were written against the BufferedReaderExample class but then, after I rewrote some example from BufReadExample in a unit-test-like style I moved them to BufReadExTest because they are now tests (although they aren't testing BufReadExample).. Would it be fine if a put them in a new BufferedReaderTest class in the same package? Or shall I rather puth them back into BufReadExample (than we will have a mix of normal example methods and unit-test methods)?